libraries/helpers.rb

```diff
@@ -45,6 +45,12 @@ def alarm_criteria_remote_http
     module ParsedParams
       include RackspaceMonitoringCookbook::Helpers::AlarmCriteria
 
+      def parsed_agent_filename
```

Comment:
Not a big change, but this could probably even be shortened to:
```
def parsed_agent_filename
  new_resource.agent_filename || new_resource.type
end
```